### PR TITLE
python312Packages.nutils: unbreak by adding missing deps; python312Packages.nutils-poly: init

### PR DESCRIPTION
**Pull request by TomaSajt**
## Description of changes

CC: @Scriptkiddi 
Saw that this package was failing on hydra, so I tried to fix it.

This PR fixes the `nutils` python package's derivation by adding required dependencies.

The `nutils-poly` package is also a new dependency, but was not yet packaged in nixpkgs.
I added this package in a separate commit.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/nutils-poly/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  dependencies = [ numpy ];
+
+  nativeCheckInputs = [ unittestCheckHook ];
+
+  pythonImportsCheck = [ "nutils_poly" ];
+
+  meta = {
+    description = "Low-level functions for evaluating and manipulating polynomials";
+    homepage = "https://github.com/nutils/poly-py";
+    license = lib.licenses.mit;
```
Comment (natsukium) [inline pkgs/development/python-modules/nutils-poly/default.nix:44]: It looks like this package doesn't have a license.
Could you ask upstream to set the license?

**Review by natsukium — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/nutils-poly/default.nix (hunk 1 above)
Comment (TomaSajt) [reply pkgs/development/python-modules/nutils-poly/default.nix:44]: Both
https://github.com/nutils/poly-py/blob/main/Cargo.toml
and
https://pypi.org/project/nutils-poly/
list the package as MIT.

Should I still ask that they add a `LICENSE` file to the repo?

Comment (natsukium) [reply pkgs/development/python-modules/nutils-poly/default.nix:44]: Thanks for letting me know. LGTM.

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 331845` at 41222b9f run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.nutils</li>
    <li><a href="https://gist.github.com/natsukium/71efa0a76487974189ad2bdda6c37077">python311Packages.nutils-poly</a></li>
    <li>python312Packages.nutils</li>
    <li><a href="https://gist.github.com/natsukium/d32ff851d60a7476622e77e7274ffc50">python312Packages.nutils-poly</a></li>
  </ul>
</details>

**Comment by TomaSajt** [comment]
I pushed a change, hopefully it fixes the error on darwin

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 331845` at d43931f3 run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>python311Packages.nutils</li>
    <li>python311Packages.nutils-poly</li>
    <li>python312Packages.nutils</li>
    <li>python312Packages.nutils-poly</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Pull request merged**